vignettes/advanced.Rmd

```diff
@@ -26,102 +29,290 @@ knitr::opts_chunk$set(
+data_ice_CIR <- data %>%
+    group_by(id = factor(id, levels = unique(id))) %>%
+    mutate(
+      is_interv = group == "Intervention",
+      has_ice = any(ind_ice1 == 1)
+    ) %>%
+    filter(is_interv & has_ice) %>% # Select pts in active arm that disc from treatment
+    summarise(
+        visit = visit[ind_ice1 == 1][1] # Select first visit affected by the ICE
+    )
+data_ice_CIR$strategy <- "CIR"
+
+# Compute change from baseline
+data$change <- data$outcome - data$outcome_bl
+
+# Remove rows corresponding to baseline visit
+data <- data[data$visit %in% levels(data$visit)[-1], ]
+data$visit <- factor(data$visit, levels <- unique(data$visit))
+
 vars <- set_vars(
-    outcome = "outcome",
-    visit = "visit",
     subjid = "id",
+    visit = "visit",
+    outcome = "change",
     group = "group",
-    covariates = c("age", "sex", "group*visit")
+    covariates = c("visit*outcome_bl", "visit*group"),
+    strategy = "strategy"
+    )
+
+vars_an <- vars
+vars_an$covariates <- "outcome_bl"
+
+```
+
+Let's assume that the chosen multiple imputation (MI) method is the approximate Bayesian MI with 10 imputations that can be set with the function `method_approxbayes()` as follows:
+
+```{r}
+method <- method_approxbayes(n_sample = 10)
+```
+
+We can now call sequentially the 4 key functions of `rbmi` to perform the multiple imputation.
+Please note that the management of post-discontinuation observed data is performed without additional complexity for the user.
+`draws()` automatically excludes post-ICE data handled with a reference-based method using information provided by the argument `data_ice`.
+`impute()` will impute only actual missing data in `data[[vars$outcome]]`.
+
+```{r}
+set.seed(123)
+draw_obj <- draws(
+    data = data,
+    data_ice = data_ice_CIR,
+    vars = vars,
+    method = method
+)
+
+impute_obj_CIR <- impute(
+    draw_obj,
+    references = c("Control" = "Control", "Intervention" = "Intervention")
 )
+
+ana_obj_CIR <- analyse(
+    impute_obj_CIR,
+    vars = vars_an
+)
+
+pool(ana_obj_CIR)
 ```
 
+# Adding time-varying covariates in the imputation model
+
+`rbmi` allows to include time-varying covariates in the imputation model.
+For example, lets suppose that we want to implement an estimator equivalent to the one described in section \@ref(sec:postICEobs) with the following differences:
+
+- There is an additional time-varying covariate that represents the number of visits from the ICE treatment discontinuation. This is set equal to 0 up to the treatment discontinuation and to the time from treatment discontinuation
+at subsequent visits. Additionally a treatment by time-varying covariate interaction terms is included in the model. This variable allows to estimate the post-ICE trajectory in each arm directly from the data, without explicitly making apriori any (reference-based) assumption.
+- The imputation is performed under MAR.
```

Comment:
This just looks a bit weird given you have 1 super long sentance and 1 short one both in bullet points,  perhaps just have two short bullet points and extract the qualifying text to a paragraph beneath it ?

Reply:
Ok, I separated the long sentence into 2 bullet points